Storybook code render

ethanWallace opened the pull request:
# Summary | Résumé

Modifications to improve code preview in storybook.

- Add `replace()` to templates to remove whitespace
- Add logic to hide component properties if they are showing the default prop value. Required fields that require user t enter information will always show.
- Modify CSS to help with rendering error and disabled states.

melaniebmn commented on packages/web/src/components/gcds-fieldset/gcds-fieldset.stories.tsx:
```diff
@@ -56,16 +56,16 @@ export default {
-  required="${args.required}"
-  disabled="${args.disabled}"
-  lang="${args.lang}"
+  ${args.hint ? `hint="${args.hint}"` : null}
```
The hint is not required, but it shows up by default. Is that on purpose to encourage people to use a hint?

melaniebmn commented on packages/web/src/components/gcds-file-uploader/gcds-file-uploader.stories.tsx:
```diff
@@ -62,48 +62,48 @@ export default {
-  accept="${args.accept}"
-  multiple="${args.multiple}"
-  lang="${args.lang}"
+  ${args.hint ? `hint="${args.hint}"` : null}
```
Same question about the hint: is it set by default on purpose to encourage people to use it?

melaniebmn commented on packages/web/src/components/gcds-file-uploader/gcds-file-uploader.stories.tsx:
```diff
@@ -62,48 +62,48 @@ export default {
 // Web Component (Angular, Vue)
 <gcds-file-uploader
   uploader-id="${args.uploaderId}"
   label="${args.label}"
```
Label is not marked as required in the story and it also doesn't have a data type defined. Do you mind adding that in?

melaniebmn commented on packages/web/src/components/gcds-file-uploader/gcds-file-uploader.stories.tsx:
```diff
@@ -62,48 +62,48 @@ export default {
+  ${args.errorMessage ? `error-message="${args.errorMessage}"` : null}
+  ${args.required ? `required` : null}
+  ${args.disabled ? `disabled` : null}
+  ${args.value ? `value="${args.value}"` : null}
```
Value is also missing the data type in the story, any chance you can add that in as well?

melaniebmn commented on packages/web/src/components/gcds-footer/gcds-footer.stories.tsx:
```diff
@@ -46,27 +46,27 @@ export default {
-  contextual-links='${args.contextualLinks}'
-  sub-links='${args.subLinks}'
-  lang="${args.lang}"
+  ${args.display != "compact" ? `display="${args.display}"` : null}
```
Similar to the button type, the footer display is required but it doesn't show up by default. Since it has a default value, does this prop still need to be required?

melaniebmn commented on packages/web/src/components/gcds-input/gcds-input.stories.tsx:
```diff
@@ -79,45 +79,45 @@ export default {
-  hide-label="${args.hideLabel}"
-  lang="${args.lang}"
+  ${args.type != "text" ? `type="${args.type}"` : null}
+  ${args.hint ? `hint="${args.hint}"` : null}
```
Same question about the hint.

melaniebmn commented on packages/web/src/components/gcds-input/gcds-input.stories.tsx:
```diff
@@ -79,45 +79,45 @@ export default {
 // Web Component (Angular, Vue)
 <gcds-input
   input-id="${args.inputId}"
   label="${args.label}"
```
Same comment about the label: would you mind marking it as required in the story and setting a default type?

melaniebmn commented on packages/web/src/components/gcds-input/gcds-input.stories.tsx:
```diff
@@ -79,45 +79,45 @@ export default {
+  ${args.errorMessage ? `error-message="${args.errorMessage}"` : null}
+  ${args.required ? `required` : null}
+  ${args.disabled ? `disabled` : null}
+  ${args.value ? `value="${args.value}"` : null}
```
Same comment about the value: would you mind setting a default type?

melaniebmn commented on packages/web/src/components/gcds-radio/gcds-radio.stories.tsx:
```diff
@@ -68,18 +68,18 @@ export default {
 // Web Component (Angular, Vue)
 <gcds-radio
   radio-id="${args.radioId}"
   label="${args.label}"
```
Same comment about the label: would you mind marking it as required in the story and setting a default type?

melaniebmn commented on packages/web/src/components/gcds-radio/gcds-radio.stories.tsx:
```diff
@@ -68,18 +68,18 @@ export default {
-  value="${args.value}"
-  checked="${args.checked}"
-  lang="${args.lang}"
+  ${args.hint ? `hint="${args.hint}"` : null}
```
Same question abut the hint.

melaniebmn commented on packages/web/src/components/gcds-radio/gcds-radio.stories.tsx:
```diff
@@ -68,18 +68,18 @@ export default {
+  ${args.hint ? `hint="${args.hint}"` : null}
+  ${args.required ? `required` : null}
+  ${args.disabled ? `disabled` : null}
+  ${args.value ? `value="${args.value}"` : null}
```
Same comment about the value: would you mind setting a default data type?

melaniebmn commented on packages/web/src/components/gcds-select/gcds-select.stories.tsx:
```diff
@@ -56,18 +56,18 @@ export default {
 // Web Component (Angular, Vue)
 <gcds-select
   select-id="${args.selectId}"
   label="${args.label}"
```
Same comment about the label: would you mind marking it as required in the story and setting a default data type?

melaniebmn commented on packages/web/src/components/gcds-select/gcds-select.stories.tsx:
```diff
@@ -56,18 +56,18 @@ export default {
-  default-value="${args.defaultValue}"
-  lang="${args.lang}"
+  ${args.hint ? `hint="${args.hint}"` : null}
+  ${args.errorMessage ? `error-message="${args.errorMessage}"` : null}
```
Would you mind setting a default data type for the error message?

melaniebmn commented on packages/web/src/components/gcds-select/gcds-select.stories.tsx:
```diff
@@ -56,18 +56,18 @@ export default {
+  ${args.errorMessage ? `error-message="${args.errorMessage}"` : null}
+  ${args.required ? `required` : null}
+  ${args.disabled ? `disabled` : null}
+  ${args.value ? `value="${args.value}"` : null}
```
Same comment about the value: would you mind setting a default data type?

melaniebmn commented on packages/web/src/components/gcds-textarea/gcds-textarea.stories.tsx:
```diff
@@ -63,41 +63,41 @@ export default {
 // Web Component (Angular, Vue)
 <gcds-textarea
   textarea-id="${args.textareaId}"
   label="${args.label}"
```
Same comment about the label: would you mind marking it as required in the story and setting a default data type?

melaniebmn commented on packages/web/src/components/gcds-textarea/gcds-textarea.stories.tsx:
```diff
@@ -63,41 +63,41 @@ export default {
+  ${args.errorMessage ? `error-message="${args.errorMessage}"` : null}
+  ${args.required ? `required` : null}
+  ${args.disabled ? `disabled` : null}
+  ${args.value ? `value="${args.value}"` : null}
```
Same comment about the value: would you mind setting a default data type?

melaniebmn commented on packages/web/src/components/gcds-textarea/gcds-textarea.stories.tsx:
```diff
@@ -63,41 +63,41 @@ export default {
-  hide-label="${args.hideLabel}"
-  lang="${args.lang}"
+  ${args.hint ? `hint="${args.hint}"` : null}
+  ${args.errorMessage ? `error-message="${args.errorMessage}"` : null}
```
Same comment about the error message: would you mind setting a default data type?

ethanWallace commented:
@melaniebmn Removed unneeded default values and fixed attributes rendering in properties table. Removed `...formProps` from stories for now since they would be hit or miss when loading property values properly.

melaniebmn reviewed (changes requested):
Looks good! One more change: can you remove the button type as required from the component itself so that users don't run into any issues?

ethanWallace commented:
@melaniebmn Seems the button type property is already set to not required

melaniebmn reviewed (approved):
Oops sorry, must have misread that yesterday. Looks good to merge then :)
